### PR TITLE
Added CVE-2020-1752

### PR DESCRIPTION
**Pull request by tausif-rh**

### CODE REVIEW

File: 2020/1xxx/CVE-2020-1752.json [hunk 1]
```diff
@@ -4,15 +4,68 @@
...
+        ]
+    },
+    "references": {
+        "reference_data": [
+            {
+                "url": "https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2020-1752",
+                "name": "https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2020-1752",
+                "refsource": "CONFIRM"
+            }
+        ]
```
Comment (iamleot) [inline 2020/1xxx/CVE-2020-1752.json:51]: Please add upstream bug report and commit:

 https://sourceware.org/bugzilla/show_bug.cgi?id=25414
 https://sourceware.org/git/gitweb.cgi?p=glibc.git;h=ddc650e9b3dc916eab417ce9f79e67337b05035c

File: 2020/1xxx/CVE-2020-1752.json [hunk 2]
```diff
@@ -4,15 +4,68 @@
...
     },
     "description": {
         "description_data": [
             {
                 "lang": "eng",
-                "value": "** RESERVED ** This candidate has been reserved by an organization or individual that will use it when announcing a new security problem. When the candidate has been publicized, the details for this candidate will be provided."
+                "value": "A use-after-free vulnerability introduced in upstream version 2.14 was found glibc in the way the tilde expansion was carried out. Directory paths containing an initial tilde followed by a valid username were affected by this issue. A local attacker could exploit this flaw by creating a specially crafted path that, when processed by the glob function, would potentially lead to arbitrary code execution."
```
Comment (iamleot) [inline 2020/1xxx/CVE-2020-1752.json:57]: Please mention the fixed version here too.

File: 2020/1xxx/CVE-2020-1752.json [hunk 3]
```diff
@@ -4,15 +4,68 @@
...
+                {
+                    "vendor_name": "GNU Libc",
+                    "product": {
+                        "product_data": [
+                            {
+                                "product_name": "glibc",
+                                "version": {
+                                    "version_data": [
+                                        {
+                                            "version_value": "Fixed in 2.29"
```
Comment (iamleot) [inline 2020/1xxx/CVE-2020-1752.json:21]: glibc-2.29 seems that was released on 2019-01-31 but the corresponding commit that fix that issue landed in corresponding master on 2020-02-20.  I think that need to be adjusted to `2.32` and eventually update it if possible further (older) releases will contain that.

If that's not the case, can you please describe that further?

Comment (tausif-rh) [reply 2020/1xxx/CVE-2020-1752.json:21]: Working on this.

Comment (tausif-rh) [reply 2020/1xxx/CVE-2020-1752.json]: The patch was cherry-picked in branches 'release/2.29/master', 'release/2.30/master' and 'release/2.31/master' from master commit ddc650e9b3dc916eab417ce9f79e67337b05035c. That's why I set Fixed in: glibc 2.29.

```
$ git log --grep="Fix use-after-free in glob" origin/release/2.29/master
commit 9aaebaf805f24ae10e0bfad332d6d5eabd58c451
Author: Andreas Schwab <schwab@suse.de>
Date:   Wed Feb 19 17:21:46 2020 +0100

    Fix use-after-free in glob when expanding ~user (bug 25414)
    
    The value of `end_name' points into the value of `dirname', thus don't
    deallocate the latter before the last use of the former.
    
    (cherry picked from commit ddc650e9b3dc916eab417ce9f79e67337b05035c)
```

Note that those branches are not "tagged", meaning that you won't find them in `git log -l`. I've noticed that 'release/2.26/master' and 'release/2.28/master' now also have the fix, which was not backported to those branches when I first analyzed the flaw. For some reason 'release/2.27/master' does not contain the fix, though.

We're waiting for the patch to land in the next glibc-2.32. I've removed it from the fixed in for now.

File: 2020/1xxx/CVE-2020-1752.json (hunk 2 above)
Comment (tausif-rh) [reply 2020/1xxx/CVE-2020-1752.json]: Updated the description to signify affected version. For fixed in version, please see me resolving comment below.

File: 2020/1xxx/CVE-2020-1752.json (hunk 3 above)
Comment (iamleot) [reply 2020/1xxx/CVE-2020-1752.json]: Given that it's already present in the NEWS file I would put there 2.32 anyways (also if not released).

File: 2020/1xxx/CVE-2020-1752.json (hunk 1 above)
Comment (iamleot) [reply 2020/1xxx/CVE-2020-1752.json:51]: This is not resolved.

**Pull request merged**